server/start.go

```diff
@@ -542,6 +519,7 @@ func startAPIServer(ctx context.Context, g *errgroup.Group, cmtCfg *cmtcfg.Confi
 	}
 	// TODO: Why do we reload and unmarshal the entire genesis doc in order to get the chain ID.
 	// surely theres a better way. This is likely a serious node start time overhead.
+	// Shouldn't it be in cmtCfg.ChainID() ?
```

Comment:
the genesis and config can have the incorrect values if a user is syncing from a snapshot, im not sure if this is needed

Reply:
I see, RIP

Reply:
Should we move this into a follow-up issue for `StartAPIServer: how to get chain-id` ?